Pfappserver Updates

satkunas opened the pull request:
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Delete branch after merge
YES

cgx commented on html/pfappserver/root/static.alt/package.json:
```diff
@@ -21,6 +21,7 @@
     "vue-browser-acl": "^0.5.5",
     "vue-i18n": "^7.8.1",
     "vue-router": "^3.0.1",
+    "vue2vis": "0.0.13",
```
I guess it would be safe to use the caret range `"^0.0.x"` (>=0.0.0 < 0.1.0).

cgx commented on html/pfappserver/root/static.alt/src/utils/network.js:
```diff
@@ -0,0 +1,37 @@
+      is8021X: false,
+      isMacAuth: false
+    }
+    if (cType.search(new RegExp(/inline/i)) !== -1) {
```
For the same result, I personally prefer the following syntax:

```js
if (/inline/i.test(cType)) {
```

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -142,14 +230,20 @@
-            {{ $t('Option82') }}
+            {{ $t('Option82') }} <b-badge pill v-if="node.dhcpoption82 && node.dhcpoption82.length > 0" variant="light" class="ml-1">{{ node.dhcpoption82.length }}</b-badge>
           </template>
+          <b-table stacked="sm" :items="node.dhcpoption82" :fields="dhcpOption82Fields" v-if="node.dhcpoption82" striped></b-table>
```
We should probably create a new component for *dhcpoption82* since we have a searchable API endpoint.

satkunas replied:
Agreed, I'll defer on writing the component for now and will revisit this again later.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -19,84 +19,172 @@
+              </pf-form-row>
+              <pf-form-row class="text-nowrap" :label="$t('Last Connection Sub Type')">
+                {{ node.last_connection_sub_type }}
+              </pf-form-row>
```
We should combine the display of the connection type and subtype.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -19,84 +19,172 @@
+              <pf-form-row class="text-nowrap" :label="$t('Last .1X Username')">
+                {{ node.last_dot1x_username }}
+              </pf-form-row>
+              <pf-form-row class="text-nowrap" :label="$t('Last Port')">
```
We should combine the display of the port, switch and switch MAC.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -19,84 +19,172 @@
+              </pf-form-row>
+              <pf-form-row class="text-nowrap" :label="$t('Last Start Timestamp')">
+                {{ node.last_start_timestamp }}
+              </pf-form-row>
```
Is the *start time* and *start timestamp* redundant information?

satkunas replied:
Not sure, I haven't seen any live example with these fields populated.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -19,84 +19,172 @@
+                {{ node.regdate }}
+              </pf-form-row>
+              <pf-form-row class="text-nowrap" :label="$t('Unregistration Date')">
+                {{ node.unregdate }}
```
*Unregistration date* should be editable.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -284,6 +439,39 @@ export default {
+        // noop
+      }).catch(() => {
+        // noop
+      })
```
Remove both `.then` and `.catch` calls.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -284,6 +439,39 @@ export default {
+      }).catch(() => {
+        // noop
+      })
+    },
```
Remove both `.then` and `.catch` calls.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue:
```diff
@@ -284,6 +439,39 @@ export default {
+      }).catch(() => {
+        // noop
+      })
+    },
```
Remove both `.then` and `.catch` calls.

satkunas replied:
I'll expand these both when I have a better idea how a successful API response looks.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js:
```diff
@@ -339,6 +354,48 @@ const actions = {
+    return new Promise((resolve, reject) => {
+      api.reevaluateAccessNode(data).then(response => {
+        if (response.status === 'success') {
+          // noop
```
Missing some logic here?

satkunas replied:
I'll expand these both when I have a better idea how a successful API response looks.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js:
```diff
@@ -339,6 +354,48 @@ const actions = {
+    return new Promise((resolve, reject) => {
+      api.refreshFingerbankNode(data).then(response => {
+        if (response.status === 'success') {
+          // noop
```
Missing some logic here?

satkunas replied:
I'll expand these both when I have a better idea how a successful API response looks.

cgx commented on html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js:
```diff
@@ -339,6 +354,48 @@ const actions = {
+    return new Promise((resolve, reject) => {
+      api.restartSwitchportNode(data).then(response => {
+        if (response.status === 'success') {
+          // noop
```
Missing some logic here?

satkunas replied:
I'll expand these both when I have a better idea how a successful API response looks.

cgx commented on html/pfappserver/root/static.alt/src/views/Users/_components/UsersSearch.vue:
```diff
@@ -86,10 +86,160 @@ export default {
         },
+        {
+          value: 'title',
+          text: 'Title',
```
Every `text` attribute of all *fields* should be translated.

satkunas commented:
@cgx I've addressed all issues that are currently relevant. I'll follow up with the remaining issues at a later time. Feel free to merge this into `devel` anytime.
